src/cascade/executor/dismod_runner.py

```diff
@@ -153,51 +162,46 @@ def run_and_watch(command, single_use_machine, poll_time):
         str: The output stream.
         str: The error stream.
     """
+    command = [str(a) for a in command]
+    loop = asyncio.get_event_loop()
+    return loop.run_until_complete(_async_run_and_watch(command, single_use_machine, poll_time))
+
+
+@asyncio.coroutine
+def _async_run_and_watch(command, single_use_machine, poll_time):
     if single_use_machine:
         pre_execution_function = reduce_process_priority
     else:
         pre_execution_function = None
 
     try:
         CODELOG.info(f"Forking to {command}")
-        sub_process = subprocess.Popen(
-            args=command,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            universal_newlines=True,
-            preexec_fn=pre_execution_function
-         )
-    except ValueError as ve:
-        raise Exception(
-            f"Dismod called with invalid arguments {ve}"
+        sub_process = yield from asyncio.subprocess.create_subprocess_exec(
+            *command, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE, preexec_fn=pre_execution_function
         )
+    except ValueError as ve:
+        raise Exception(f"Dismod called with invalid arguments {ve}")
     except OSError as ose:
-        raise Exception(
-            f"Dismod couldn't run due to OS error {ose}"
-        )
-
-    out_list = list()
-    err_list = list()
+        raise Exception(f"Dismod couldn't run due to OS error {ose}")
 
-    while sub_process.poll() is not None:
-        outs, errs = sub_process.communicate(timeout=poll_time)
-        MATHLOG.debug(outs)
-        out_list.append(outs)
-        err_list.append(errs)
+    out_list = []
+    err_list = []
 
-    outs, errs = sub_process.communicate()
-    MATHLOG.debug(outs)
-    out_list.append(outs)
-    err_list.append(errs)
+    loop = asyncio.get_event_loop()
+    std_out_task = loop.create_task(_read_pipe(sub_process.stdout, out_list, lambda text: MATHLOG.debug(text)))
+    std_err_task = loop.create_task(_read_pipe(sub_process.stderr, err_list, lambda text: MATHLOG.error(text)))
```

Comment:
Separate for stderr and stdout. Makes sense. This code is pretty compact. If it passes tests, then I'm good with this level of complexity.